internal/dataplane/config_status.go

```diff
@@ -25,25 +32,39 @@ type NoOpConfigStatusNotifier struct{}
 
 var _ ConfigStatusNotifier = NoOpConfigStatusNotifier{}
 
-func (n NoOpConfigStatusNotifier) NotifyConfigStatus(status ConfigStatus) {
+func (n NoOpConfigStatusNotifier) NotifyConfigStatus(_ context.Context, _ ConfigStatus) {
 }
 
 type ChannelConfigNotifier struct {
-	ch chan ConfigStatus
+	ch     chan ConfigStatus
+	logger logr.Logger
 }
 
 var _ ConfigStatusNotifier = &ChannelConfigNotifier{}
 
-func (n *ChannelConfigNotifier) NotifyConfigStatus(status ConfigStatus) {
-	n.ch <- status
+// NotifyConfigStatus sends the status in a separate goroutine. If the notification is not received in 1s, it's dropped.
+func (n *ChannelConfigNotifier) NotifyConfigStatus(ctx context.Context, status ConfigStatus) {
+	const notifyTimeout = time.Second
+
+	go func() {
+		select {
+		case n.ch <- status:
+		case <-ctx.Done():
+			n.logger.Info("Context done, not notifying config status", "status", status)
+		case <-time.After(notifyTimeout):
```

Comment:
`time.After` "leaks" the timer. Better to use `time.NewTimer()` with deferred close

Reply:
https://github.com/Kong/kubernetes-ingress-controller/pull/3624